lib/project_types/script/layers/infrastructure/languages/assemblyscript_project_creator.rb

```diff
@@ -5,18 +5,16 @@ module Layers
     module Infrastructure
       module Languages
         class AssemblyScriptProjectCreator < ProjectCreator
-          MIN_NODE_VERSION = "14.5.0" # kept because task_runner uses this
-          NPM_SET_REGISTRY_COMMAND = "npm --userconfig ./.npmrc config set @shopify:registry https://registry.npmjs.com"
-          NPM_SET_ENGINE_STRICT_COMMAND = "npm --userconfig ./.npmrc config set engine-strict true"
-
           def self.config_file
             "package.json"
           end
 
           def setup_dependencies
+            task_runner = Infrastructure::Languages::AssemblyScriptTaskRunner.new(ctx)
+            task_runner.ensure_environment
+
             super
-            command_runner.call(NPM_SET_REGISTRY_COMMAND)
-            command_runner.call(NPM_SET_ENGINE_STRICT_COMMAND)
+            task_runner.set_npm_config
```

Comment:
Any reason this has to be done after the super and not before, with the other task_runner call? Not a big deal, just thinking about code prose

Reply:
I don't think there's a reason for it. It was just how it was setup before so I kept it in that order. That being said if I move it up, it could treat that as the ensure_env check as well 🤔